compiler/rustc_target/src/callconv/aarch64.rs

```diff
@@ -27,18 +32,40 @@ where
 
         let valid_unit = match unit.kind {
             RegKind::Integer => false,
-            RegKind::Float => true,
+            RegKind::Float => cx.target_spec().abi != "softfloat",
             RegKind::Vector => size.bits() == 64 || size.bits() == 128,
         };
 
         valid_unit.then_some(Uniform::consecutive(unit, size))
     })
 }
 
+fn softfloat_float_abi<Ty>(target: &Target, arg: &mut ArgAbi<'_, Ty>) {
+    if target.abi != "softfloat" {
+        return;
+    }
+    // Do *not* use the floag registers for passing arguments, as that would make
+    // the ABI depend on whether `neon` instructions are enabled.
+    // Apparently there is no standard ABI here [1], so we can do whatever we want.
```

Comment:
Apologies if I missed discussion of this in the related issues but I think there _is_ a standard ABI: AAPCS-soft. It specifies that floats should be passed as equivalently sized integers so you're doing the right thing here. 

I think though we should reword the related comments to refer to the appropriate docs. 
- ARM ABI Docs: https://github.com/ARM-software/abi-aa/pull/232
- Clang implementation: https://github.com/llvm/llvm-project/pull/84146

Reply:
I was just echoing what other people told me -- specifically @pinskia. They then added "But it is only for the R cores rather than the A cores"; I don't know what that means.

Reply:
>  It specifies that floats should be passed as equivalently sized integers so you're doing the right thing here.

I am not sure I am doing the right thing for the `is_homogeneous_aggregate` logic. Presumably there, instead of treating a struct of floats as non-homogeneous, we want to treat it like a homogeneous sequence of integers?

Reply:
Ah, but integers are not a "valid unit" so it's fine.

Reply:
I have updated the comments to match this. I have also made the Rust ABI for pairs involving floats less dumb, avoiding a pointer indirection when the data fits into a pointer-sized register.

Reply:
Hmm... that's true. It looks like there are two "profiles" of hardware: A and R. The A profile mandates support for FP at the hardware level so there isn't an officially defined ABI for softfloat there. The R profile makes FP support optional and now defines the AAPCS-soft ABI. 

I think it would be valuable to mention _something_ about AAPCS-soft here even if it's not wholly applicable just to show there is some precedent for what we're doing.

Reply:
Although maybe it would make more sense to change this target to properly use the R profile? Then we would actually be using the correct ABI...

Reply:
Some people need softfloat on hardware that has float support. A prominent example is the Linux kernel. So I don't think we want to mandate the use of hardfloats even for the A profile.